Frontend/src/Components/Dashboard/GraphContainer.js

```diff
@@ -51,6 +51,47 @@ export default function GraphContainer({ queue, latestTime, ...props }) {
   // stores the list of graph names/id
   const [graphTitles, setGraphTitles] = useState(["", "", ""]);
 
+  const [savedGraphsMetadata, setSaveGraphsMetadata] = useState({}) //(NOTE: not linked to render) This will save saved graph titles and corresponding data keys
+  const [neededGraphMetadata, setNeededGraphMetadata] = useState({ meta: [] })
+
+  const sendNGM_toBackend = async (data) => {
+    let response = await fetch("http://localhost:4001/api/needed-graph-metadata", {
+      method: 'POST', // *GET, POST, PUT, DELETE, etc.
+      // mode: 'cors', // no-cors, *cors, same-origin
+      headers: {
+        'Accept': 'application/json',
+        'Content-type': 'application/json; charset=UTF-8',
+      },
+      body: JSON.stringify(data) // body data type must match "Content-Type"
+    })
+
+    console.log("SENDING NGM:", data)
+
+    console.log(response)
+  }
+
+  const updateSavedGraphsMetadata = useCallback((graphName, graphKeys) => {
+    if ((graphName !== "Custom" || graphName !== "") && graphName) {
```

Comment:
```suggestion
    if ((graphName !== "Custom" && graphName !== "") && graphName) {
```
If you meant to check for graphName not being "Custom" or empty